src/plugins/data/public/antlr/promql/code_completion.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { monaco } from '@osd/monaco';
+import { CursorPosition, PromQLAutocompleteResult } from '../shared/types';
+import { openSearchPromQLAutocompleteData } from './promql_autocomplete';
+import { QuerySuggestion, QuerySuggestionGetFnArgs } from '../../autocomplete';
+import { parseQuery } from '../shared/utils';
+import { SuggestionItemDetailsTags } from '../shared/constants';
+
+export interface SuggestionParams {
+  position: monaco.Position;
+  query: string;
+}
+
+export interface ISuggestionItem {
+  text: string;
+  type: string;
+  fieldType?: string;
+}
+
+export const getSuggestions = async ({
+  selectionStart,
+  selectionEnd,
+  indexPattern,
+  dataset,
+  position,
+  query,
+  services,
+}: QuerySuggestionGetFnArgs): Promise<QuerySuggestion[]> => {
+  if (!services || !services.appName || !indexPattern || !dataset) return [];
+  try {
+    const { lineNumber, column } = position || {};
+    const suggestions = getOpenSearchPromQLAutoCompleteSuggestions(query, {
+      line: lineNumber || selectionStart,
+      column: column || selectionEnd,
+    });
+    const prometheusResourceClient = services.data.resourceClientFactory.create('prometheus');
+
+    const finalSuggestions: QuerySuggestion[] = [];
+
+    if (suggestions.suggestMetrics) {
+      const metrics = await prometheusResourceClient.getMetricMetadata(dataset.id);
+      finalSuggestions.push(
+        ...Object.keys(metrics).map((af) => ({
+          text: `${af}`,
+          type: monaco.languages.CompletionItemKind.Method,
+          detail: SuggestionItemDetailsTags.AggregateFunction,
+        }))
+      );
+    }
+
+    if (suggestions.suggestLabels || suggestions.suggestLabels === '') {
+      const labels = await prometheusResourceClient.getLabels(
+        dataset.id,
+        suggestions.suggestLabels !== '' ? suggestions.suggestLabels : undefined
+      );
+      finalSuggestions.push(
+        ...labels.map((af: string) => ({
+          text: `${af}`,
+          type: monaco.languages.CompletionItemKind.File,
+          detail: SuggestionItemDetailsTags.Table,
+        }))
+      );
+    }
+
+    if (suggestions.suggestLabelValues && suggestions.suggestLabelValues.label) {
+      // TODO: update when we can get metric name passed in
```

Comment:
For my context, in the meantime is this suggesting the same labels for all metrics?

Reply:
This is for suggesting the values of a label and right now it's suggesting all values as if we had a label matcher without any metric.
In the conditional above, we're able to suggest the only relevant labels for a metric.